src/App.tsx

```diff
@@ -16,35 +26,121 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([...goodsFromServer]);
+  const [sortState, setSortState] = useState<{
+    reversed: boolean;
+    criterion: SortType;
+  }>({
+    reversed: false,
+    criterion: SortType.None,
+  });
+  const [isReset, setIsReset] = useState(false as boolean);
+
+  const sortGoodsByAlphabet: ClickButton = () => {
+    const sortedGoods: Good[] = [...goods].sort((a, b) => {
+      if (sortState.reversed) {
+        return b.localeCompare(a);
+      }
+
+      return a.localeCompare(b);
+    });
+
+    setGoods(sortedGoods);
+    setSortState({ ...sortState, criterion: SortType.Alphabet });
+    setIsReset(true);
+  };
+
+  const sortGoodsByLength: ClickButton = () => {
+    const sortedGoods = [...goods].sort((a, b) => {
+      if (sortState.reversed) {
+        return b.length - a.length;
+      }
+
+      return a.length - b.length;
+    });
+
+    setGoods(sortedGoods);
+    setSortState({ ...sortState, criterion: SortType.Length });
+    setIsReset(true);
+  };
+
+  const reverseGoods: ClickButton = () => {
+    setSortState(prevState => {
+      const reversed = !prevState.reversed;
+
+      setGoods(prevGoods => {
+        const newGoods = [...prevGoods].reverse();
+
+        setIsReset(
+          !newGoods.every(
+            (element, index) => element === goodsFromServer[index],
+          ),
```

Comment:
The logic for setting `isReset` in the `reverseGoods` function might be misleading. It checks if the reversed goods match the original order, which might not be the intended behavior. Consider revisiting this logic to ensure it aligns with the intended functionality.